src/Sulu/Bundle/AdminBundle/Resources/js/components/ImageSectorSelection/ImageSectorSelection.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+import {action, computed, observable} from 'mobx';
+import DimensionsConverter from './DimensionsConverter';
+import React from 'react';
+import RectangleSelection from '../RectangleSelection';
+import RoundingNormalizer from '../RectangleSelection/dataNormalizers/RoundingNormalizer';
+import type {SelectionData} from '../RectangleSelection/types';
+import log from 'loglevel';
+import {observer} from 'mobx-react';
+import selectionStyles from './imageSectorSelection.scss';
+
+@observer
+export default class ImageSectorSelection extends React.PureComponent {
+    props: {
+        /** Determines the position at which the selection box is rendered at the beginning. */
+        initialSelection?: SelectionData,
+        minWidth?: number,
+        minHeight?: number,
+        onChange?: (s: SelectionData) => void,
+        imageSrc: string,
+    };
+
+    image: Image;
+    rounding = new RoundingNormalizer();
+
+    @observable imageLoaded = false;
+    @observable containerWidth: number;
+    @observable containerHeight: number;
+
+    @action componentWillMount() {
+        this.image = new Image();
+        this.image.onload = action(() => this.imageLoaded = true);
+        this.image.onerror = () => log.error('Failed to preload image');
+        this.image.src = this.props.imageSrc;
+    }
+
+    @computed get converter(): DimensionsConverter {
+        return new DimensionsConverter(
+            this.image.naturalWidth, this.image.naturalHeight, this.imageResizedWidth, this.imageResizedHeight);
+    }
+
+    @computed get imageResizedHeight(): number {
+        if (this.imageTouchesHoriziontalBorders()) {
+            return Math.min(this.image.naturalHeight, this.containerHeight);
+        } else {
+            return this.imageResizedWidth * this.image.naturalHeight / this.image.naturalWidth;
+        }
+    }
+
+    @computed get imageResizedWidth(): number {
+        if (this.imageTouchesHoriziontalBorders()) {
+            return this.imageResizedHeight * this.image.naturalWidth / this.image.naturalHeight;
+        } else {
+            return Math.min(this.image.naturalWidth, this.containerWidth);
+        }
+    }
+
+    imageTouchesHoriziontalBorders() {
```

Comment:
This method is not responsible for normalizing or sanitizing the component's data in any way. This is just a helper method used to calculate the correct `imageResizedWidth` and `imageResizedHeight`.